enterprise/views.py

```diff
@@ -1198,57 +1112,66 @@ def get_enterprise_program_enrollment_page(
             program_details['weeks_to_complete_max'],
         )
 
-        if program_enrollment_details['program_price'] != program_enrollment_details['program_discounted_price']:
-            discount_provider = self.context_data['discount_provider'].format(
-                strong_start='<strong>',
-                strong_end='</strong>',
-                provider=enterprise_customer.name,
-            )
-        else:
-            discount_provider = None
-
-        if program_enrollment_details['enrolled_in_program']:
+        # Update some enrollment-related text requirements.
+        if program_details['enrolled_in_program']:
             purchase_action = self.actions['purchase_unenrolled_courses']
             item = self.items['enrollment']
         else:
             purchase_action = self.actions['purchase_program']
             item = self.items['program_enrollment']
 
+        # Add any warning messages.
+        program_data_sharing_consent = get_data_sharing_consent(
+            request.user.username,
+            enterprise_customer.uuid,
+            program_uuid=program_details['uuid'],
+        )
+        if program_data_sharing_consent.exists and not program_data_sharing_consent.granted:
+            add_consent_declined_message(request, enterprise_customer, program_title)
+
+        discount_data = program_details['discount_data']
+        if discount_data.get('total_incl_tax_excl_discounts') is None:
+            add_missing_price_information_message(request, program_title)
+
+        one_click_purchase_eligibility = program_details['is_learner_eligible_for_one_click_purchase']
+        if not one_click_purchase_eligibility:
+            add_not_one_click_purchasable_message(request, enterprise_customer, program_title)
```

Comment:
Sorry for moving code around, like here. Since I was refactoring, I tried to make it more organized, as well.